x/stakeibc/keeper/msg_server_submit_tx.go

```diff
@@ -181,49 +181,147 @@ func (k Keeper) ProcessUpdateUndelegatedBalance(ctx sdk.Context) {
+						}
+					} else {
+						k.Logger(ctx).Info(fmt.Sprintf("Did NOT set redemption rate at H=%d because no controller balances", hz.DelegationAccount.HeightLastQueriedDelegatedBalance))
+					}
+				}
+				k.Logger(ctx).Info(fmt.Sprintf("Did NOT set redemption rate at H=%d because last UB and DB update heights didn't match.", hz.DelegationAccount.HeightLastQueriedDelegatedBalance))
 
-			ctx.EventManager().EmitEvents(sdk.Events{
-				sdk.NewEvent(
-					sdk.EventTypeMessage,
-					sdk.NewAttribute("hostZone", zoneInfo.ChainId),
-					sdk.NewAttribute("totalUndelegatedBalance", balance.String()),
-				),
-			})
+			} else {
+				k.Logger(ctx).Info(fmt.Sprintf("Opted to NOT set HeightLastQueriedUndelegatedBalance because query height %d is less than last update's height %d.", h, da.HeightLastQueriedUndelegatedBalance))
+			}
 			// ---------------------------------
 
 			return nil
 		}
-		k.Logger(ctx).Info(fmt.Sprintf("\tQuerying balance for %s", zoneInfo.ChainId))
-		k.InterchainQueryKeeper.QueryBalances(ctx, zoneInfo, queryBalanceCB, delegationIca.Address)
+		k.Logger(ctx).Info(fmt.Sprintf("\tQuerying UndelegatedBalance for %s at %d height", zoneInfo.ChainId, height))
+		k.InterchainQueryKeeper.QueryBalances(ctx, zoneInfo, queryUndelegatedBalanceCB, delegationIca.Address, height)
 		return nil
 	}
 
 	// Iterate the zones and apply updateUndelegatedBal
 	k.IterateHostZones(ctx, updateUndelegatedBal)
 }
 
+func (k Keeper) ProcessUpdateBalances(ctx sdk.Context, height int64) {
+	// TODO(now) ; rm this after debug; should never call at 0!
+	if height == 0 {
+		return
+	}
+	updateBalances := func(ctx sdk.Context, index int64, zoneInfo types.HostZone) error {
+		k.Logger(ctx).Info(fmt.Sprintf("\tUpdating balances on %s", zoneInfo.ChainId))
+
+		delegationIca := zoneInfo.GetDelegationAccount()
+		if delegationIca == nil || delegationIca.Address == "" {
+			k.Logger(ctx).Error("Zone %s is missing a delegation address!", zoneInfo.ChainId)
+			return errors.New("Zone is missing a delegation address!")
+		}
+		var queryDBCBDaisyChain icqkeeper.Callback = func(icqk icqkeeper.Keeper, ctx sdk.Context, args []byte, query icqtypes.Query, h int64) error {
+			k.Logger(ctx).Info(fmt.Sprintf("\tdelegation DelegatedBalance callback on %s", zoneInfo.HostDenom))
+			var response stakingTypes.QueryDelegatorDelegationsResponse
+			err := k.cdc.Unmarshal(args, &response)
+			if err != nil {
+				k.Logger(ctx).Error("Unable to unmarshal balances info for zone", "err", err)
+				return err
+			}
+
+			// Add up delegations across validators
+			delegatorSum := sdk.NewCoin(zoneInfo.HostDenom, sdk.ZeroInt())
+			for _, delegation := range response.DelegationResponses {
+				delegatorSum = delegatorSum.Add(delegation.Balance)
+				if err != nil {
+					return err
+				}
+			}
+			// --- Update Undelegated Balance ---
```

Comment:
Should this read "Update Delegated Balance"? Possible I'm missing something but it seems like we're updating the delegated balance in the following lines